src/components/YearlyDataTab.tsx

```diff
@@ -26,26 +37,56 @@ export default class YearlyDataTab extends React.Component<YearlyDataTabProps, Y
         let year = library.basic_info.timestamp && library.basic_info.timestamp.match(/20\d+/)[0];
         if (year) {
           if (sortedByYear[year]) {
-            sortedByYear[year][catName].push(library.basic_info.name);
+            sortedByYear[year][catName].push(library);
           } else {
             sortedByYear[year] = {"production": [], "testing": [], "cancelled": []};
-            sortedByYear[year][catName].push(library.basic_info.name);
+            sortedByYear[year][catName].push(library);
           }
         }
       });
     });
-    return(sortedByYear);
+    return sortedByYear;
   }
   toggleFormatting() {
     this.setState({ styled: !this.state.styled });
   }
+  getYearlyTotal(data: {[key: string]: LibraryData[]}) {
+    return Object.values(data).map(v => v.length).reduce((accum, next) => accum + next);
+  }
+  toggleExpanded(e) {
+    let [verb, year] = e.target.textContent.toLowerCase().split(" ");
+    let yearsToShow = {};
+    let newValue = verb === "show";
+    if (year === "all") {
+      Object.keys(this.state.yearsToShow).forEach(year => yearsToShow[year] = newValue);
+    } else {
+      yearsToShow[year] = newValue;
+      yearsToShow = { ...this.state.yearsToShow, ...yearsToShow };
+    }
+    this.setState({...this.state, ...{ yearsToShow }});
+  }
   render(): JSX.Element {
-    let categories = (year) => <ul>{Object.keys(year).map(category => <li className={this.state.styled ? "stats-category" : ""}><p className="stats-category-name">{category.substr(0, 1).toUpperCase() + category.substr(1)}</p>{names(year[category])}</li>)}</ul>;
-    let names = (category) => <ul className="yearly-library-list">{category.map(l => <li>{l}</li>)}</ul>;
     let sortedByYear = this.sortByYear(this.props.data);
-    let years = Object.keys(sortedByYear).map(y =>
-      <li key={y} className={this.state.styled ? "year-li" : ""}><p>{y}</p>{categories(sortedByYear[y])}</li>
-    );
+    let showOrHideAll: string = `${Object.keys(sortedByYear).every(year => this.state.yearsToShow[year]) ? "Hide" : "Show"} All`;
+    let showOrHideYear = (year) => `${this.state.yearsToShow[year] ? "Hide" : "Show"} ${year}`;
+    let total = Object.keys(sortedByYear).map(y => this.getYearlyTotal(sortedByYear[y]));
+    let years = Object.keys(sortedByYear).map((y) => {
+      let yearlyTotal = this.getYearlyTotal(sortedByYear[y]);
```

Comment:
All the yearly totals are in `total`. I think you can do `let yearlyTotal = total[index]` and in the map do `(y, index)`. Just to make one less call to `getYearlyTotal`.